cmd/collectors/zapi/plugins/snapmirror/snapmirror.go

```diff
@@ -246,3 +209,56 @@ func (my *SnapMirror) updateLimitCache() error {
 	my.Logger.Debug().Msgf("updated limit cache for %d nodes", count)
 	return nil
 }
+
+func (my *SnapMirror) updateProtectedfields(instance *matrix.Instance) {
+
+	// check for group_type
+	if instance.GetLabel("group_type") != "" {
+		groupType := instance.GetLabel("group_type")
+		destinationVolume := instance.GetLabel("destination_volume")
+		sourceVolume := instance.GetLabel("source_volume")
+		destinationLocation := instance.GetLabel("destination_location")
+
+		isSvmDr := groupType == "vserver" && destinationVolume == "" && sourceVolume == ""
+		isCg := groupType == "CONSISTENCYGROUP" && strings.Contains(destinationLocation, ":/cg/")
+		isConstituentVolumeRelationshipWithinSvmDr := groupType == "vserver" && !strings.HasSuffix(destinationLocation, ":")
+		isConstituentVolumeRelationshipWithinCG := groupType == "CONSISTENCYGROUP" && !strings.Contains(destinationLocation, ":/cg/")
+
+		// Update protectedBy field
+		if isSvmDr || isConstituentVolumeRelationshipWithinSvmDr {
+			instance.SetLabel("protectedBy", "storage_vm")
+		} else if isCg || isConstituentVolumeRelationshipWithinCG {
+			instance.SetLabel("protectedBy", "cg")
+		} else {
+			instance.SetLabel("protectedBy", "volume")
+		}
+
+		// SVM-DR related information is populated
+		// Update protectionSourceType field
+		if isSvmDr {
```

Comment:
thought so, but it would again need nested if/else as isSvmDr and isConstituentVolumeRelationshipWithinSvmDr would follow different route, same apply for cg.